src/prime_cli/commands/env.py

```diff
@@ -381,8 +381,87 @@ def push(
                         f"{owner_info['name']}/{env_name}[/green]"
                     )
             except APIError as e:
-                console.print(f"[red]Failed to resolve environment: {e}[/red]")
-                raise typer.Exit(1)
+                # Handle missing username (slug) by prompting user to set it and retrying
+                err_msg = str(e)
+                if "missing a username" in err_msg.lower():
+                    console.print(
+                        "[yellow]Your user profile is missing a username.[/yellow] "
+                        "You must choose a username to publish environments."
+                    )
+                    console.print(
+                        "[dim]Note: This username can only be chosen once and will be public.[/dim]"
+                    )
+
+                    while True:
+                        try:
+                            chosen = (
+                                typer.prompt(
+                                    "Enter your desired username",
+                                )
+                                .strip()
+                                .lower()
+                            )
+                        except typer.Abort:
+                            console.print("[red]Cancelled by user[/red]")
+                            raise typer.Exit(1)
+
+                        if not chosen:
+                            console.print("[red]Username cannot be empty[/red]")
+                            continue
+
+                        if not re.match(r"^[a-z0-9-]{3,30}$", chosen):
+                            console.print(
+                                "[red]Invalid username.[/red] "
+                                "Use 3-30 chars with lowercase letters, numbers, and '-' only."
+                            )
+                            continue
+
+                        try:
+                            client.patch("/users/slug", json={"slug": chosen})
```

Comment:
This looks like the wrong route
```suggestion
                            client.patch("/user/slug", json={"slug": chosen})
```